### PR TITLE
Unify system shutdown, reboot and app cleanup routines

### PR DESCRIPTION
**Pull request by jpnurmi**
Share the common functionality, needed for several pages, with a `SystemShutdown` mixin available in `ubuntu_wizard` utils.

Close: #224

### CODE REVIEW

File: packages/ubuntu_desktop_installer/lib/pages/turn_off_bitlocker/turn_off_bitlocker_model.dart [hunk 1]
```diff
@@ -2,14 +2,15 @@ import 'package:subiquity_client/subiquity_client.dart';
...
+  TurnOffBitLockerModel(this.client);
 
-  final SubiquityClient _client;
+  @override
+  final SubiquityClient client;
 
-  /// Requests system reboot.
-  Future<void> reboot() {
-    return _client.reboot(immediate: true).then((_) => closeWindow());
+  /// Requests an immediate system reboot.
```
Comment (robert-ancell) [inline packages/ubuntu_desktop_installer/lib/pages/turn_off_bitlocker/turn_off_bitlocker_model.dart:12]: Comments seems misleading - inly immediate if `immediate=true, right?

File: packages/ubuntu_desktop_installer/lib/pages/turn_off_rst/turn_off_rst_model.dart [hunk 2]
```diff
@@ -2,14 +2,15 @@ import 'package:subiquity_client/subiquity_client.dart';
...
+  TurnOffRSTModel(this.client);
 
-  final SubiquityClient _client;
+  @override
+  final SubiquityClient client;
 
-  /// Requests system reboot.
-  Future<void> reboot() {
-    return _client.reboot(immediate: true).then((_) => closeWindow());
+  /// Requests an immediate system reboot.
```
Comment (robert-ancell) [inline packages/ubuntu_desktop_installer/lib/pages/turn_off_rst/turn_off_rst_model.dart:12]: Same issue as previous comment.

**Review by robert-ancell — APPROVED** [review]
LGTM

**Comment by jpnurmi** [comment]
Thanks for the reviews!

**Pull request merged**